pkgs/applications/office/watson/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv, pythonPackages }:
+
+pythonPackages.buildPythonApplication rec {
+  pname = "td-watson";
+  name = "${pname}-${version}";
+  version = "1.4.0";
+
+  src = pythonPackages.fetchPypi {
+    inherit version pname;
+    sha256 = "1py0g4990jmvq0dn7jasda7f10kzr41bix46hnbyc1rshjzc17hq";
+  };
+
+  # uses tox, test invocation fails
+  doCheck = false;
+  #checkPhase = ''
+  #  tox $out
```

Comment:
Thanks, good to know. I'd like to avoid that kind of work. 
Apparently doing checks/tests of Python applications in nixpkgs is not trivial.

Here's the output of `checkPhase`

```
running install tests
flake8 create: /tmp/nix-build-td-watson-1.4.0.drv-0/td-watson-1.4.0/.tox/flake8
flake8 installdeps: flake8
flake8 develop-inst: /tmp/nix-build-td-watson-1.4.0.drv-0/td-watson-1.4.0
flake8 installed: Error [Errno 2] No such file or directory: 'git' while executing command git rev-parse,The directory '/homeless-shelter/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.,appdirs==1.4.3,arrow==0.10.0,click==6.7,flake8==3.3.0,mccabe==0.6.1,packaging==16.8,pycodestyle==2.3.1,pyflakes==1.5.0,pyparsing==2.2.0,python-dateutil==2.6.0,requests==2.13.0,six==1.10.0,td-watson==1.4.0
flake8 runtests: PYTHONHASHSEED='1534061741'
flake8 runtests: commands[0] | flake8 --show-source watson/ tests/ scripts/
py27 create: /tmp/nix-build-td-watson-1.4.0.drv-0/td-watson-1.4.0/.tox/py27
ERROR: InterpreterNotFound: python2.7
py33 create: /tmp/nix-build-td-watson-1.4.0.drv-0/td-watson-1.4.0/.tox/py33
ERROR: InterpreterNotFound: python3.3
py34 create: /tmp/nix-build-td-watson-1.4.0.drv-0/td-watson-1.4.0/.tox/py34
ERROR: InterpreterNotFound: python3.4
py35 create: /tmp/nix-build-td-watson-1.4.0.drv-0/td-watson-1.4.0/.tox/py35
ERROR: invocation failed (exit code 1), logfile: /tmp/nix-build-td-watson-1.4.0.drv-0/td-watson-1.4.0/.tox/py35/log/py35-0.log
ERROR: actionid: py35
msg: getenv
cmdargs: ['/nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/bin/python3.5m', '-m', 'virtualenv', '--python', '/nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/bin/python3.5', 'py35']
env: {'out': '/nix/store/58m79slqfyy54b4wbzplhjiwxc5yv4rv-td-watson-1.4.0', 'PWD': '/tmp/nix-build-td-watson-1.4.0.drv-0/td-watson-1.4.0', 'name': 'td-watson-1.4.0', 'TEMP': '/tmp/nix-build-td-watson-1.4.0.drv-0', 'NIX_CFLAGS_COMPILE': ' -isystem /nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/include', 'propagatedBuildInputs': '/nix/store/qwk4zlzkypwwk28y5a55p823z0rdkzn4-python3.5-requests-2.13.0 /nix/store/hrxj8bz3m0xdbs8kdv5zgyz2386a7gl5-python3.5-click-6.7 /nix/store/gyrvq8n0rk1s8hrsg11hz1jh1qqlv0v3-python3.5-arrow-0.7.0 /nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3 /nix/store/4m77il34i2bdzqzlrnl3wxnf2ib2rbdr-python3.5-setuptools-30.2.0', 'shell': '/nix/store/53h800j8kgpj0a349f7wxa5hgkj1vby2-bash-4.4-p12/bin/bash', 'NIX_STORE': '/nix/store', 'OLDPWD': '/tmp/nix-build-td-watson-1.4.0.drv-0/td-watson-1.4.0/dist', 'installPhase': 'runHook preInstall\n\nmkdir -p "$out/lib/python3.5/site-packages"\nexport PYTHONPATH="$out/lib/python3.5/site-packages:$PYTHONPATH"\n\npushd dist\n/nix/store/q6vady3xiski46vkd6mgf1vm10wnrnid-python3.5-bootstrapped-pip-9.0.1/bin/pip install *.whl --no-index --prefix=$out --no-cache  --build tmpbuild\npopd\n\nrunHook postInstall\n', 'doInstallCheck': '1', 'NIX_CC': '/nix/store/a7why3dakidpf8sywipvdq52969cwzqy-gcc-wrapper-5.4.0', 'doCheck': '', 'NIX_BUILD_CORES': '3', 'checkPhase': 'tox $out\n', 'installCheckPhase': 'tox $out\n', 'nativeBuildInputs': '', 'SOURCE_DATE_EPOCH': '1478024239', 'namePrefix': '', 'system': 'x86_64-linux', 'stdenv': '/nix/store/wbvjr5m7nwkwb9pzxwqq3w8kgrhnpcwi-stdenv', 'propagatedNativeBuildInputs': '', 'src': '/nix/store/fnfnqql0056h0axa59nqz0qrjxgmvlzh-td-watson-1.4.0.tar.gz', 'TZ': 'UTC', 'CC': 'gcc', 'SHELL': '/nix/store/53h800j8kgpj0a349f7wxa5hgkj1vby2-bash-4.4-p12/bin/bash', 'CONFIG_SHELL': '/nix/store/53h800j8kgpj0a349f7wxa5hgkj1vby2-bash-4.4-p12/bin/bash', 'NIX_ENFORCE_PURITY': '1', 'pythonPath': '', 'NIX_ENFORCE_NO_NATIVE': '1', 'VIRTUAL_ENV': '/tmp/nix-build-td-watson-1.4.0.drv-0/td-watson-1.4.0/.tox/py35', 'TEMPDIR': '/tmp/nix-build-td-watson-1.4.0.drv-0', 'builder': '/nix/store/53h800j8kgpj0a349f7wxa5hgkj1vby2-bash-4.4-p12/bin/bash', 'PYTHONHASHSEED': '1534061741', 'pname': 'td-watson', 'postFixup': 'wrapPythonPrograms\n# Check if we have two packages with the same name in the closure and fail.\n# If this happens, something went wrong with the dependencies specs.\n# Intentionally kept in a subdirectory, see catch_conflicts/README.md.\n/nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/bin/python3.5m /nix/store/lvg31h3na663n390jysm22q3z4lsbc3b-catch_conflicts/catch_conflicts.py\n', 'buildInputs': '/nix/store/hwrzivkc0n2j7m6hrf8srg8dgcj9mpjl-hook /nix/store/q6vady3xiski46vkd6mgf1vm10wnrnid-python3.5-bootstrapped-pip-9.0.1 /nix/store/s58kc90ijqsyzn9nssi7xkn58ckbmvnq-hook /nix/store/j9834hmkkhy5ic1lfz6msfwrvhz74fzj-python3.5-tox-2.4.1 /nix/store/mzkxpwjhbkla2jqq3pn9pr8n56193hc0-python3.5-pytest-3.0.7 /nix/store/6fywxrlazbnsbb1ql5pdj41sg5sayqix-python3.5-pytest-mock-1.2 /nix/store/lvljwn2hnl2xsc2hd65dz8pv5qy0m6iw-python3.5-pytest-runner-2.6.2 /nix/store/8qnsq4ia87rlz5ah6ysny7zgf784dw63-python3.5-flake8-3.2.1 /nix/store/fmpjrw2f9ga4nr836dc5nacn9q2vdnrh-python3.5-virtualenv-15.0.3', 'TMP': '/tmp/nix-build-td-watson-1.4.0.drv-0', 'configurePhase': 'runHook preConfigure\nrunHook postConfigure\n', 'CXX': 'g++', 'NIX_LDFLAGS': '-rpath /nix/store/58m79slqfyy54b4wbzplhjiwxc5yv4rv-td-watson-1.4.0/lib64 -rpath /nix/store/58m79slqfyy54b4wbzplhjiwxc5yv4rv-td-watson-1.4.0/lib  -L/nix/store/q6vady3xiski46vkd6mgf1vm10wnrnid-python3.5-bootstrapped-pip-9.0.1/lib -L/nix/store/j9834hmkkhy5ic1lfz6msfwrvhz74fzj-python3.5-tox-2.4.1/lib -L/nix/store/4apnrriwrmv8x6650d2rsj8xk44k1v6c-python3.5-py-1.4.31/lib -L/nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/lib -L/nix/store/4m77il34i2bdzqzlrnl3wxnf2ib2rbdr-python3.5-setuptools-30.2.0/lib -L/nix/store/fmpjrw2f9ga4nr836dc5nacn9q2vdnrh-python3.5-virtualenv-15.0.3/lib -L/nix/store/0plg9qywh95nh4mfvzq3h8yl25g4ib4f-python3.5-pluggy-0.3.1/lib -L/nix/store/mzkxpwjhbkla2jqq3pn9pr8n56193hc0-python3.5-pytest-3.0.7/lib -L/nix/store/6fywxrlazbnsbb1ql5pdj41sg5sayqix-python3.5-pytest-mock-1.2/lib -L/nix/store/czsqr46r68ngbkjcxnsl3jd8cbirxd53-python3.5-mock-2.0.0/lib -L/nix/store/njlr9zrmq4zq5g6bjf02i484pbqxsdjh-python3.5-funcsigs-1.0.2/lib -L/nix/store/6b7rv0nphdcpxnqbf8w937g1kf9yz9vh-python3.5-six-1.10.0/lib -L/nix/store/isipl9p1s4y2g6id7dy2mzxcsgnnssfi-python3.5-pbr-1.8.1/lib -L/nix/store/lvljwn2hnl2xsc2hd65dz8pv5qy0m6iw-python3.5-pytest-runner-2.6.2/lib -L/nix/store/8qnsq4ia87rlz5ah6ysny7zgf784dw63-python3.5-flake8-3.2.1/lib -L/nix/store/2znqcxbf9mpc9bza07fvx9v5f98y6kwv-python3.5-pyflakes-1.3.0/lib -L/nix/store/z3j2q69xbrk7b5h17xcsxcb7iaqg61s8-python3.5-pycodestyle-2.0.0/lib -L/nix/store/ggd7ls1pbzg1s5alh2zgp6djgfgg2nfy-python3.5-mccabe-0.5.3/lib -L/nix/store/qwk4zlzkypwwk28y5a55p823z0rdkzn4-python3.5-requests-2.13.0/lib -L/nix/store/hrxj8bz3m0xdbs8kdv5zgyz2386a7gl5-python3.5-click-6.7/lib -L/nix/store/gyrvq8n0rk1s8hrsg11hz1jh1qqlv0v3-python3.5-arrow-0.7.0/lib -L/nix/store/8rk21rkhnn7f1i7k0kibw4ygxmm738g0-python3.5-dateutil-2.6.0/lib', 'PATH': '/tmp/nix-build-td-watson-1.4.0.drv-0/td-watson-1.4.0/.tox/py35/bin:/nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/bin:/nix/store/j9834hmkkhy5ic1lfz6msfwrvhz74fzj-python3.5-tox-2.4.1/bin:/nix/store/4m77il34i2bdzqzlrnl3wxnf2ib2rbdr-python3.5-setuptools-30.2.0/bin:/nix/store/fmpjrw2f9ga4nr836dc5nacn9q2vdnrh-python3.5-virtualenv-15.0.3/bin:/nix/store/q6vady3xiski46vkd6mgf1vm10wnrnid-python3.5-bootstrapped-pip-9.0.1/bin:/nix/store/j9834hmkkhy5ic1lfz6msfwrvhz74fzj-python3.5-tox-2.4.1/bin:/nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/bin:/nix/store/4m77il34i2bdzqzlrnl3wxnf2ib2rbdr-python3.5-setuptools-30.2.0/bin:/nix/store/fmpjrw2f9ga4nr836dc5nacn9q2vdnrh-python3.5-virtualenv-15.0.3/bin:/nix/store/mzkxpwjhbkla2jqq3pn9pr8n56193hc0-python3.5-pytest-3.0.7/bin:/nix/store/isipl9p1s4y2g6id7dy2mzxcsgnnssfi-python3.5-pbr-1.8.1/bin:/nix/store/8qnsq4ia87rlz5ah6ysny7zgf784dw63-python3.5-flake8-3.2.1/bin:/nix/store/2znqcxbf9mpc9bza07fvx9v5f98y6kwv-python3.5-pyflakes-1.3.0/bin:/nix/store/z3j2q69xbrk7b5h17xcsxcb7iaqg61s8-python3.5-pycodestyle-2.0.0/bin:/nix/store/rya0fnyl5mq32yd7caciq8hgrv85hkhs-patchelf-0.9/bin:/nix/store/k9vd0cpzbssf69i0gkd2sxmrcp2p1dkr-paxctl-0.9/bin:/nix/store/a7why3dakidpf8sywipvdq52969cwzqy-gcc-wrapper-5.4.0/bin:/nix/store/n2c3cqjhaa2qbhxhx266rvq2j4wjb1mv-gcc-5.4.0/bin:/nix/store/2xrkq62pm03bwqympr1nldqfni658qhf-binutils-2.28/bin:/nix/store/la68p6a78mg1w2w1pai2d29g06sjgng1-glibc-2.25-bin/bin:/nix/store/d56snr32kff2l1941jknkpf076h6mbwf-coreutils-8.27/bin:/nix/store/d56snr32kff2l1941jknkpf076h6mbwf-coreutils-8.27/bin:/nix/store/4r8sm9439yzfwyisb9f6jxcgz3ciqh1h-findutils-4.6.0/bin:/nix/store/vxsrk1c0ghm458lhcj8sig8wl2vgjnmb-diffutils-3.5/bin:/nix/store/1f5iq0qp412dpzqzz6shyij1hh1m9svl-gnused-4.4/bin:/nix/store/839jqxk1x5d5v2fwcfb7g6x1cd329a7g-gnugrep-3.0/bin:/nix/store/nh235jq6m3hh27cypb3kkfyclbhz8vdy-gawk-4.1.4/bin:/nix/store/s1ncq0376pa75xylzxqh6w89krw5nv7b-gnutar-1.29/bin:/nix/store/rz6wkw3sj85qq8sjzqlhaj4zplx3k3av-gzip-1.8/bin:/nix/store/dv59hggw1dkh2ihbifpn27rhk6c6mj1s-bzip2-1.0.6.0.1-bin/bin:/nix/store/lq7a1qwiqv80yxwqvb5l6sfgq8744mzj-gnumake-4.2.1/bin:/nix/store/53h800j8kgpj0a349f7wxa5hgkj1vby2-bash-4.4-p12/bin:/nix/store/455x0g35hfavb4yk3nbdavp1s23z5jva-patch-2.7.5/bin:/nix/store/v6mvhsdrn38bjvx3xlkizb9rhi1pspb2-xz-5.2.3-bin/bin', 'DETERMINISTIC_BUILD': '1', 'shellHook': '\nif test -e setup.py; then\n  tmp_path=$(mktemp -d)\n  export PATH="$tmp_path/bin:$PATH"\n  export PYTHONPATH="$tmp_path/lib/python3.5/site-packages:$PYTHONPATH"\n  mkdir -p $tmp_path/lib/python3.5/site-packages\n  /nix/store/q6vady3xiski46vkd6mgf1vm10wnrnid-python3.5-bootstrapped-pip-9.0.1/bin/pip install -e . --prefix $tmp_path >&2\nfi\n\n', 'TMPDIR': '/tmp/nix-build-td-watson-1.4.0.drv-0', 'SSL_CERT_FILE': '/no-cert-file.crt', 'program_PYTHONPATH': '/nix/store/58m79slqfyy54b4wbzplhjiwxc5yv4rv-td-watson-1.4.0/lib/python3.5/site-packages:/nix/store/qwk4zlzkypwwk28y5a55p823z0rdkzn4-python3.5-requests-2.13.0/lib/python3.5/site-packages:/nix/store/4m77il34i2bdzqzlrnl3wxnf2ib2rbdr-python3.5-setuptools-30.2.0/lib/python3.5/site-packages:/nix/store/hrxj8bz3m0xdbs8kdv5zgyz2386a7gl5-python3.5-click-6.7/lib/python3.5/site-packages:/nix/store/gyrvq8n0rk1s8hrsg11hz1jh1qqlv0v3-python3.5-arrow-0.7.0/lib/python3.5/site-packages:/nix/store/8rk21rkhnn7f1i7k0kibw4ygxmm738g0-python3.5-dateutil-2.6.0/lib/python3.5/site-packages:/nix/store/6b7rv0nphdcpxnqbf8w937g1kf9yz9vh-python3.5-six-1.10.0/lib/python3.5/site-packages', 'version': '1.4.0', 'checkInputs': '/nix/store/j9834hmkkhy5ic1lfz6msfwrvhz74fzj-python3.5-tox-2.4.1 /nix/store/mzkxpwjhbkla2jqq3pn9pr8n56193hc0-python3.5-pytest-3.0.7 /nix/store/6fywxrlazbnsbb1ql5pdj41sg5sayqix-python3.5-pytest-mock-1.2 /nix/store/lvljwn2hnl2xsc2hd65dz8pv5qy0m6iw-python3.5-pytest-runner-2.6.2 /nix/store/8qnsq4ia87rlz5ah6ysny7zgf784dw63-python3.5-flake8-3.2.1 /nix/store/fmpjrw2f9ga4nr836dc5nacn9q2vdnrh-python3.5-virtualenv-15.0.3', '_PATH': '/nix/store/q6vady3xiski46vkd6mgf1vm10wnrnid-python3.5-bootstrapped-pip-9.0.1/bin:/nix/store/j9834hmkkhy5ic1lfz6msfwrvhz74fzj-python3.5-tox-2.4.1/bin:/nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/bin:/nix/store/4m77il34i2bdzqzlrnl3wxnf2ib2rbdr-python3.5-setuptools-30.2.0/bin:/nix/store/fmpjrw2f9ga4nr836dc5nacn9q2vdnrh-python3.5-virtualenv-15.0.3/bin:/nix/store/mzkxpwjhbkla2jqq3pn9pr8n56193hc0-python3.5-pytest-3.0.7/bin:/nix/store/isipl9p1s4y2g6id7dy2mzxcsgnnssfi-python3.5-pbr-1.8.1/bin:/nix/store/8qnsq4ia87rlz5ah6ysny7zgf784dw63-python3.5-flake8-3.2.1/bin:/nix/store/2znqcxbf9mpc9bza07fvx9v5f98y6kwv-python3.5-pyflakes-1.3.0/bin:/nix/store/z3j2q69xbrk7b5h17xcsxcb7iaqg61s8-python3.5-pycodestyle-2.0.0/bin:/nix/store/rya0fnyl5mq32yd7caciq8hgrv85hkhs-patchelf-0.9/bin:/nix/store/k9vd0cpzbssf69i0gkd2sxmrcp2p1dkr-paxctl-0.9/bin:/nix/store/a7why3dakidpf8sywipvdq52969cwzqy-gcc-wrapper-5.4.0/bin:/nix/store/n2c3cqjhaa2qbhxhx266rvq2j4wjb1mv-gcc-5.4.0/bin:/nix/store/2xrkq62pm03bwqympr1nldqfni658qhf-binutils-2.28/bin:/nix/store/la68p6a78mg1w2w1pai2d29g06sjgng1-glibc-2.25-bin/bin:/nix/store/d56snr32kff2l1941jknkpf076h6mbwf-coreutils-8.27/bin', 'buildPhase': 'runHook preBuild\ncp /nix/store/fscd8f71wmpwphcmi5mx8qnif2402x9m-run_setup.py nix_run_setup.py\n/nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/bin/python3.5m nix_run_setup.py  bdist_wheel\nrunHook postBuild\n', 'NIX_BUILD_TOP': '/tmp/nix-build-td-watson-1.4.0.drv-0', 'SHLVL': '1', 'NIX_INDENT_MAKE': '1', 'program_PATH': '/nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/bin:/nix/store/58m79slqfyy54b4wbzplhjiwxc5yv4rv-td-watson-1.4.0/bin:/nix/store/4m77il34i2bdzqzlrnl3wxnf2ib2rbdr-python3.5-setuptools-30.2.0/bin', 'HOME': '/homeless-shelter'}

/nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/bin/python3.5m: No module named virtualenv

ERROR: InvocationError: /nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/bin/python3.5m -m virtualenv --python /nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/bin/python3.5 py35 (see /tmp/nix-build-td-watson-1.4.0.drv-0/td-watson-1.4.0/.tox/py35/log/py35-0.log)
___________________________________ summary ____________________________________
  flake8: commands succeeded
SKIPPED:  py27: InterpreterNotFound: python2.7
SKIPPED:  py33: InterpreterNotFound: python3.3
SKIPPED:  py34: InterpreterNotFound: python3.4
ERROR:   py35: InvocationError: /nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/bin/python3.5m -m virtualenv --python /nix/store/7ifgcl36dl34nd8cy1c4pgzz9fa9n61f-python3-3.5.3/bin/python3.5 py35 (see /tmp/nix-build-td-watson-1.4.0.drv-0/td-watson-1.4.0/.tox/py35/log/py35-0.log)
builder for ‘/nix/store/20i8n0va9j9w0zpgl02iqajsa15kcf90-td-watson-1.4.0.drv’ failed with exit code 1
error: build of ‘/nix/store/20i8n0va9j9w0zpgl02iqajsa15kcf90-td-watson-1.4.0.drv’ failed

```